src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+    const length = this.length;
+
+    for (let i = 0; i < this.length; i++) {
+      if (typeof this[i] === 'undefined'
+      || this[i] === null) {
+        this[i] = '';
+      }
+
+      if (i !== length - 1) {
+        string += this[i] + String(separator);
+      } else {
+        string += this[i];
+      }
```

Comment:
and then you'd only add separator here like that
```suggestion
      if (i !== length - 1) {
        string += String(separator);
      } 
```